src/platform_impl/linux/x11/mod.rs

```diff
@@ -573,14 +571,14 @@ impl<T> EventLoopWindowTarget<T> {
         &self.xconn
     }
 
-    pub fn set_device_event_filter(&self, filter: DeviceEventFilter) {
-        self.device_event_filter.set(filter);
+    pub fn set_listen_device_events(&self, allowed: DeviceEvents) {
+        self.device_events.set(allowed);
     }
 
     /// Update the device event filter based on window focus.
-    pub fn update_device_event_filter(&self, focus: bool) {
-        let filter_events = self.device_event_filter.get() == DeviceEventFilter::Never
-            || (self.device_event_filter.get() == DeviceEventFilter::Unfocused && !focus);
+    pub fn update_listen_device_events(&self, focus: bool) {
+        let filter_events = self.device_events.get() == DeviceEvents::Never
+            || (self.device_events.get() == DeviceEvents::WhenFocused && !focus);
 
         let mut mask = 0;
         if !filter_events {
```

Comment:
Could you reverse all of that to `Always` and remove `filter` word all together?

Reply:
What do you mean revert all of that to 'Always'?

Reply:
So we don't have to `!filter_events`? It would be `let device_events = self.device_events.get() == DeviceEvents::Always || (focus && self.device.events.get() == DeviceEvents::WhenFocused)` and then `if device_events`.